Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -2771,6 +2771,45 @@ bool CSSPropertyParser::consumeWhiteSpaceShorthand(bool important)
     return true;
 }
 
+
+bool CSSPropertyParser::consumeAnimationRangeShorthand(bool important)
+{
+    // TODO: handle animation range sequences
+    RefPtr animationRangeStart = consumeAnimationRangeStart(m_range, m_context);
+    if (!animationRangeStart)
+        return false;
+
+    RefPtr<CSSValue> animationRangeEnd;
+    m_range.consumeWhitespace();
+    if (m_range.atEnd()) {
+        // From the spec: If <'animation-range-end'> is omitted and <'animation-range-start'> includes a component, then
+        // animation-range-end is set to that same and 100%. Otherwise, any omitted longhand is set to its initial value.
+
+        auto rangeEndValueForStartValue = [](const CSSValue& value) {
+            if (dynamicDowncast<CSSPrimitiveValue>(value) && SingleTimelineRange::isOffsetValue(*dynamicDowncast<CSSPrimitiveValue>(value)))
+                return CSSPrimitiveValue::create(CSSValueNormal);
+            return CSSPrimitiveValue::create(value.valueID());
+        };
+
+        if (RefPtr primitiveStart = dynamicDowncast<CSSPrimitiveValue>(animationRangeStart))
+            animationRangeEnd = rangeEndValueForStartValue(*primitiveStart);
+        else {
+            RefPtr pairStart = dynamicDowncast<CSSValuePair>(animationRangeStart);
```

Comment:
If we know the type of something then we use `downcast<>`, not `dynamicDowncast<>` plus a handwritten assert.

Reply:
Fixed.